go/cmd/dolt/commands/engine/sqlengine.go

```diff
@@ -334,6 +341,50 @@ func configureBinlogReplicaController(config *SqlEngineConfig, engine *gms.Engin
 	return nil
 }
 
+// configureBinlogPrimaryController configures the |engine| to use the default Dolt binlog primary controller, as well
+// as enabling the binlog producer if @@log_bin has been set to 1.
+//
+// NOTE: By default, binary logging for Dolt is not enabled, which differs from MySQL's @@log_bin default. Dolt's
+// binary logging is initially an opt-in feature, but we may change that after measuring and tuning the
+// performance hit that binary logging adds.
+func configureBinlogPrimaryController(engine *gms.Engine) error {
+	primaryController := dblr.NewDoltBinlogPrimaryController()
+	engine.Analyzer.Catalog.BinlogPrimaryController = primaryController
```

Comment:
It would be pretty easy to pass in the binlog primary controller instance as config to the engine/analyzer construction code in GMS, but there's additional work to turn on the binlog producer that would still need to be at this level, so I don't think it would change much currently. We could of course add a new interface or callback or something at the GMS layer to clean it up more. 

When I started looking at log file persistence, I noticed that some of this initialization needs to change anyway, so I'll take a look at cleaning up this setup code as part of that in the next PR.